packages/compiler/src/core/types.ts

```diff
@@ -1358,6 +1388,7 @@ export interface OperationStatementNode extends BaseNode, DeclarationNode, Templ
 export interface ModelStatementNode extends BaseNode, DeclarationNode, TemplateDeclarationNode {
   readonly kind: SyntaxKind.ModelStatement;
   readonly properties: readonly (ModelPropertyNode | ModelSpreadPropertyNode)[];
+  readonly propertiesRange: TextRange;
```

Comment:
i don't think we should have another range added for all of those, can this not be resolved in other ways(checking if you already have a extends, is, etc.)?

Reply:
I tried to do it based on the existing node tree but the problem here is that we need to know the position of the list start and end (in detail the pos of {}<>()), but they are ignored and do not exist in the node tree. without this information, we can't distinguish the scenario like 'model M<T|>{}' and 'model M <T>|{}' or 'scalar S | {' and 'scalar S { |' ... which should have different completion suggestions when the cursor is in or not in the scope of the list. 
Please let me know if there are any better way for me to get that information. thanks.

Reply:
actually I tried to use bodyRange at first... but then I found for scenario like model M<T> {...}, it has two different body ranges, so I change to use a more specific range name. How do you think we should name in that case? call {} bodyRange and <> templateParameterRange?

As for some added ranges not needed. I added them just in case they are useful in the future. I am fine to remove them. thx

Reply:
updated to use bodyRange and templateParametersRange and remove the other range. please help to double check. thx.

Reply:
@bterlson , do you have any comments for this change? thanks.